src/bone.js

```diff
@@ -1291,8 +1291,8 @@ class Bone {
     }
 
     for (const name in attributes) {
-      const { columnName } = attributes[name];
-      if (!(columnName in row)) {
+      const attribute = attributes[name];
+      if (!(attribute.columnName in row) && !attribute.virtual) {
```

Comment:
virtual column should not be added to unset in case of custom getter cannot be serialize
```js
class User {
 @Column()
 id: bigint;
 
 // custom will be ignored when toObject()/toJSON() execute, so make it a VIRTUAL column
 get custom(): string {
    return 'v';
 }
}
```